From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: musa <musa@local>
Date: Wed, 3 Jun 2026 11:47:36 +0000
Subject: [PATCH] MUSA: vllm.profiler.wrapper

---
 vllm/profiler/wrapper.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/vllm/profiler/wrapper.py b/vllm/profiler/wrapper.py
index 877a390..5bc3eea 100644
--- a/vllm/profiler/wrapper.py
+++ b/vllm/profiler/wrapper.py
@@ -164,11 +164,12 @@ class WorkerProfiler(ABC):
         return nullcontext()
 
 
-TorchProfilerActivity = Literal["CPU", "CUDA", "PrivateUse1", "XPU"]
+TorchProfilerActivity = Literal["CPU", "CUDA", "PrivateUse1", "MUSA", "XPU"]
 TorchProfilerActivityMap = {
     "CPU": torch.profiler.ProfilerActivity.CPU,
     "CUDA": torch.profiler.ProfilerActivity.CUDA,
     "PrivateUse1": torch.profiler.ProfilerActivity.PrivateUse1,
+    "MUSA": torch.profiler.ProfilerActivity.MUSA,
     "XPU": torch.profiler.ProfilerActivity.XPU,
 }
 
