From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: musa <musa@local>
Date: Wed, 3 Jun 2026 11:47:36 +0000
Subject: [PATCH] MUSA: vllm.v1.attention.ops.flashmla

---
 vllm/v1/attention/ops/flashmla.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vllm/v1/attention/ops/flashmla.py b/vllm/v1/attention/ops/flashmla.py
index c84a495..bbba514 100644
--- a/vllm/v1/attention/ops/flashmla.py
+++ b/vllm/v1/attention/ops/flashmla.py
@@ -55,7 +55,7 @@ def is_flashmla_dense_supported() -> tuple[bool, str | None]:
     is_available, maybe_reason = _is_flashmla_available()
     if not is_available:
         return False, maybe_reason
-    if not current_platform.is_device_capability_family(90):
+    if current_platform.get_device_capability()[0] != 3:
         return False, "FlashMLA Dense is only supported on Hopper devices."
     return True, None
 
