From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: musa <musa@local>
Date: Wed, 3 Jun 2026 11:47:36 +0000
Subject: [PATCH] MUSA: vllm.v1.worker.gpu_input_batch

---
 vllm/v1/worker/gpu_input_batch.py | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/vllm/v1/worker/gpu_input_batch.py b/vllm/v1/worker/gpu_input_batch.py
index 7585f4d..671279a 100644
--- a/vllm/v1/worker/gpu_input_batch.py
+++ b/vllm/v1/worker/gpu_input_batch.py
@@ -315,6 +315,8 @@ class InputBatch:
         self.sampled_token_ids_cpu: torch.Tensor | None = None
         self.async_copy_ready_event: torch.Event | None = None
 
+        self._num_reqs_padded: int | None = None
+
     @property
     def req_ids(self) -> list[str]:
         # None elements should only be present transiently
@@ -724,6 +726,7 @@ class InputBatch:
             self._req_ids.clear()
             self.req_output_token_ids.clear()
             self.spec_token_ids.clear()
+            self._num_reqs_padded = None
             return
 
         # NOTE(woosuk): This function assumes that the empty_req_indices
@@ -838,6 +841,8 @@ class InputBatch:
     def refresh_metadata(self):
         """Apply any batch updates to sampling metadata."""
 
+        self._num_reqs_padded = None
+
         if self.is_pooling_model:
             batch_changed = self.batch_update_builder.reset()
             if batch_changed:
@@ -1113,6 +1118,16 @@ class InputBatch:
     def num_reqs(self) -> int:
         return len(self.req_id_to_index)
 
+    @property
+    def num_reqs_padded(self) -> int:
+        if self._num_reqs_padded is None:
+            return self.num_reqs
+        return self._num_reqs_padded
+
+    @num_reqs_padded.setter
+    def num_reqs_padded(self, num_reqs_padded: int) -> None:
+        self._num_reqs_padded = num_reqs_padded
+
     @property
     def all_greedy(self) -> bool:
         return len(self.random_reqs) == 0
