From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: musa <musa@local>
Date: Wed, 3 Jun 2026 11:47:36 +0000
Subject: [PATCH] MUSA: vllm.v1.worker.gpu_worker

---
 vllm/v1/worker/gpu_worker.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vllm/v1/worker/gpu_worker.py b/vllm/v1/worker/gpu_worker.py
index a3b00aaad2..cbc689c572 100644
--- a/vllm/v1/worker/gpu_worker.py
+++ b/vllm/v1/worker/gpu_worker.py
@@ -313,7 +313,7 @@ class Worker(WorkerBase):
 
     @instrument(span_name="Init device")
     def init_device(self):
-        if self.device_config.device_type == "cuda":
+        if self.device_config.device_type in ("cuda", "musa"):
             # This env var set by Ray causes exceptions with graph building.
             os.environ.pop("NCCL_ASYNC_ERROR_HANDLING", None)
             parallel_config = self.parallel_config
@@ -1173,7 +1173,7 @@ class Worker(WorkerBase):
                         self.profiler_config,
                         worker_name=trace_name,
                         local_rank=self.local_rank,
-                        activities=["CPU", "CUDA"],
+                        activities=["CPU", "MUSA"],
                     )
                     logger.debug(
                         "Starting torch profiler with trace name: %s", trace_name
