From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: musa <musa@local>
Date: Thu, 4 Jun 2026 02:46:36 +0000
Subject: [PATCH] MUSA(csrc): csrc/cuda_compat.h

---
 csrc/cuda_compat.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/csrc/cuda_compat.h b/csrc/cuda_compat.h
index d7d589d..3893d6a 100644
--- a/csrc/cuda_compat.h
+++ b/csrc/cuda_compat.h
@@ -69,7 +69,7 @@ struct Utils {
 
 #ifndef USE_ROCM
   #define VLLM_DevFuncAttribute_SET_MaxDynamicSharedMemorySize(FUNC, VAL) \
-    cudaFuncSetAttribute(FUNC, cudaFuncAttributeMaxDynamicSharedMemorySize, VAL)
+    musaFuncSetAttribute(FUNC, musaFuncAttributeMaxDynamicSharedMemorySize, VAL)
 #else
   #define VLLM_DevFuncAttribute_SET_MaxDynamicSharedMemorySize(FUNC, VAL) \
     hipFuncSetAttribute(FUNC, hipFuncAttributeMaxDynamicSharedMemorySize, VAL)
