--- a/vllm/model_executor/layers/mamba/ops/ssd_chunk_scan.py
+++ b/vllm/model_executor/layers/mamba/ops/ssd_chunk_scan.py
@@ -304,7 +304,8 @@
         if not HAS_INITSTATES and (seq_idx != seq_idx_prev):
             # if no init states AND starting a new sequence, we need zeros
             prev_states = tl.zeros(
-                (BLOCK_SIZE_DSTATE, BLOCK_SIZE_N), dtype=C_ptr.dtype.element_ty
+                (BLOCK_SIZE_DSTATE, BLOCK_SIZE_N),
+                dtype=tl.float32 if IS_MUSA else C_ptr.dtype.element_ty
             )
         else:
             # otherwise read the previous state
@@ -318,7 +319,9 @@
                 mask=(offs_k_dstate[:, None] < dstate) & (offs_n[None, :] < hdim),
                 other=0.0,
             )
-            prev_states = prev_states.to(C_ptr.dtype.element_ty)
+            prev_states = prev_states.to(
+                tl.float32 if IS_MUSA else C_ptr.dtype.element_ty
+            )
 
         if IS_MUSA:
             acc = tl.dot(C.to(tl.float32), prev_states.to(tl.float32))
@@ -341,7 +344,8 @@
             )
             if not HAS_INITSTATES and (seq_idx != seq_idx_prev):
                 prev_states = tl.zeros(
-                    (BLOCK_SIZE_K, BLOCK_SIZE_N), dtype=C_ptr.dtype.element_ty
+                    (BLOCK_SIZE_K, BLOCK_SIZE_N),
+                    dtype=tl.float32 if IS_MUSA else C_ptr.dtype.element_ty
                 )
             else:
                 prev_states = tl.load(
@@ -350,7 +354,9 @@
                     & (offs_n[None, :] < hdim),
                     other=0.0,
                 )
-                prev_states = prev_states.to(C_ptr.dtype.element_ty)
+                prev_states = prev_states.to(
+                    tl.float32 if IS_MUSA else C_ptr.dtype.element_ty
+                )
             if IS_MUSA:
                 acc += tl.dot(C.to(tl.float32), prev_states.to(tl.float32))
             else:
